### PR TITLE
variants: Add support for nRF9160

### PR DESCRIPTION
**Pull request by ddeletic**

### CODE REVIEW

File: variants/variants.h [hunk 1]
```diff
@@ -3,19 +3,22 @@
 #pragma once
 
 #ifdef CONFIG_BOARD_ARDUINO_NANO_33_BLE
-#include <arduino_nano_33_ble_pinmap.h>
+	#include <arduino_nano_33_ble_pinmap.h>
```
Comment (soburi) [inline variants/variants.h:6]: Please do not change anything that is not related to the fix.

File: variants/nrf9160dk_nrf9160/nrf9160dk_nrf9160_pinmap.h [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+ */
+
+#ifndef INCLUDE_NRF1960DK_NRF1960_PINMAP_H
+#define INCLUDE_NRF1960DK_NRF1960_PINMAP_H
+
+#include <zephyr/drivers/gpio.h>
+#include <zephyr/drivers/i2c.h>
+#include <zephyr/kernel.h>
+
+#define NRF1960DK_NRF1960
```
Comment (soburi) [inline variants/nrf9160dk_nrf9160/nrf9160dk_nrf9160_pinmap.h:14]: I'm not recommended to include definitions unrelated to the Arduino API unless there is a specific reason.

**Review by soburi — CHANGES_REQUESTED** [review]

**Review by soburi — APPROVED** [review]

**Comment by soburi** [comment]
@DhruvaG2000 Could you take look at?

File: variants/variants.h [hunk 3]
```diff
@@ -14,6 +14,9 @@
 #ifdef CONFIG_BOARD_NRF52840DK_NRF52840
 #include "nrf52840dk_nrf52840_pinmap.h"
 #endif /* CONFIG_BOARD_NRF52840DK_NRF52840 */
+#ifdef CONFIG_BOARD_NRF9160DK_NRF9160
+	#include "nrf9160dk_nrf9160_pinmap.h"
```
Comment (DhruvaG2000) [inline variants/variants.h:18]: please follow the coding pattern, no indent necessary here.

File: variants/nrf9160dk_nrf9160/nrf9160dk_nrf9160.overlay [hunk 4]
```diff
@@ -0,0 +1,192 @@
...
+				nordic,invert;
+		};
+	};
+
+	pwm1_sleep: pwm1_sleep {
+		group1 {
+			psels = <NRF_PSEL(PWM_OUT0, 1, 11)>,
+				<NRF_PSEL(PWM_OUT1, 1, 12)>,
+				<NRF_PSEL(PWM_OUT2, 1, 13)>;
+				low-power-enable;
```
Comment (DhruvaG2000) [inline variants/nrf9160dk_nrf9160/nrf9160dk_nrf9160.overlay:175]: Again, just curious, this property seems to be some sort of equivalent of isolating I/O's during low power mode?

File: variants/nrf9160dk_nrf9160/nrf9160dk_nrf9160.overlay [hunk 5]
```diff
@@ -0,0 +1,192 @@
...
+			<&arduino_header 16 0>,
+			<&arduino_header 17 0>;
+
+		adc-pin-gpios =	    
+			<&arduino_header 0 0>,
+			<&arduino_header 1 0>,
+			<&arduino_header 2 0>,
+			<&arduino_header 3 0>,
+			<&arduino_header 4 0>,
+			<&arduino_header 5 0>;
```
Comment (DhruvaG2000) [inline variants/nrf9160dk_nrf9160/nrf9160dk_nrf9160.overlay:43]: Just curious, Were you able to use and test the arduino analog read/write API support from this project?

**Review by DhruvaG2000 — CHANGES_REQUESTED** [review]
One nit, otherwise looks okay!
Thanks alot for upstreaming this @ddeletic !

**Review by DhruvaG2000 — CHANGES_REQUESTED** [review]
In this PR as well, please follow the [Zephyr Contribution Guidelines](https://docs.zephyrproject.org/latest/contribute/guidelines.html)

Commit title should look like:
```
variants: Add support for nRF9160
```
Along with some commit message and sign off.

**Review by DhruvaG2000 — APPROVED** [review]
LGTM!

File: variants/variants.h (hunk 3 above)
Comment (DhruvaG2000) [reply variants/variants.h]: @ddeletic ,

> If you insist...
> 
> However, one of the most important aspects of software development is writing code that is easy to maintain. The first

Totally agree :)

> pre-requisite of maintainable code is that it is easy to read and understand. Now, this coding pattern
> 
> ```
> #ifdef CONFIG_BOARD_ARDUINO_NANO_33_BLE
> #include <arduino_nano_33_ble_pinmap.h>
> #endif // CONFIG_BOARD_ARDUINO_NANO_33_BLE
> #ifdef CONFIG_BOARD_ARDUINO_NANO_33_BLE_SENSE
> #include <arduino_nano_33_ble_sense_pinmap.h>
> #endif // CONFIG_BOARD_ARDUINO_NANO_33_BLE_SENSE
> #ifdef CONFIG_BOARD_ARDUINO_NANO_33_IOT
> #include <arduino_nano_33_iot_pinmap.h>
> #endif /* CONFIG_BOARD_ARDUINO_NANO_33_IOT */
> #ifdef CONFIG_BOARD_NRF52840DK_NRF52840
> #include "nrf52840dk_nrf52840_pinmap.h"
> #endif /* CONFIG_BOARD_NRF52840DK_NRF52840 */
> ...
> ```
> 
> does not fall into that category. I'm sure you'll agree that
> 
> ```
> #ifdef CONFIG_BOARD_ARDUINO_NANO_33_BLE
>     #include <arduino_nano_33_ble_pinmap.h>
> #endif // CONFIG_BOARD_ARDUINO_NANO_33_BLE
> #ifdef CONFIG_BOARD_ARDUINO_NANO_33_BLE_SENSE
>     #include <arduino_nano_33_ble_sense_pinmap.h>
> #endif // CONFIG_BOARD_ARDUINO_NANO_33_BLE_SENSE
> #ifdef CONFIG_BOARD_ARDUINO_NANO_33_IOT
>     #include <arduino_nano_33_iot_pinmap.h>
> #endif /* CONFIG_BOARD_ARDUINO_NANO_33_IOT */
> #ifdef CONFIG_BOARD_NRF52840DK_NRF52840
>     #include "nrf52840dk_nrf52840_pinmap.h"
> #endif /* CONFIG_BOARD_NRF52840DK_NRF52840 */
> ...
> ```
> 
> is much easier to follow.* 

NAK, if it were the case of simple `if()` statements you had my full support to indent. However when it comes to `#ifdef`s things are different. The general rule here is that one should write code "pretending" as though there was no #ifdef there.
They are just meant to remove / add pieces of code at compile time and play no real role in the logic itself in most cases.
Hence imagine this piece of code compiled,

```
 #ifdef CONFIG_BOARD_NRF52840DK_NRF52840
     #include "nrf52840dk_nrf52840_pinmap.h"
#endif /* CONFIG_BOARD_NRF52840DK_NRF52840 */
```

would look like:
```
     #include "nrf52840dk_nrf52840_pinmap.h"
```

The indent makes no sense in such a case right?

> You are probably thinking right now, who is this rude dude trampling over my code telling me it is no good. 

In the spirit of Open Source, I am always open to suggestions :) 

> That is definitely not my intention. But, from experience, I know that this sort of thing will never be done unless it is gradually improved, as we go along. After all, we are talking about white spaces...

As far as I have grepped in zephyr codebase, `rg -i "\#ifdef"` didn't lead me to the coding style you are suggesting above. Hence, I am only trying to stick as close as possible to what ZephyrProject follows 😄 

> 
> `*` The original pull request was written as above, with all include statements nested. However, @soburi requested that no changes are made that are not related to the pull request itself.

I agree with @soburi  here, unless there's some critical fix let's not touch code that isn't related to the purpose of this PR.

**Pull request merged**